GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                permissionObject.ResourceId = _graphAppObjectId;
+
+                //ClientId = objectId of servicePrincipal
+                permissionObject.ClientId = newServicePrincpal.ObjectId;
+
+                // add the oauth2 permission scope grant
+                try
+                {
+                    await client.Oauth2PermissionGrants.AddOAuth2PermissionGrantAsync(permissionObject);
+                    Console.WriteLine("New Permission object created: " + permissionObject.ObjectId);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError with Permission Creation: {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+
+                // remove the oauth2 permission scope grant
+                try
+                {
+                    newServicePrincpal.Oauth2PermissionGrants.Remove(permissionObject);
+                    await newServicePrincpal.UpdateAsync();
+                    Console.WriteLine("Removed Permission object: " + permissionObject.ObjectId);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError with Permission Creation: {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+
+                try
+                {
+                    await newServicePrincpal.DeleteAsync();
+                    Console.WriteLine("Deleted service principal object: " + newServicePrincpal.ObjectId);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError with Service Principal deletion: {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+
+            }
+
+            #endregion
+        }
+
+        private static async Task PrintAllPermissions(IActiveDirectoryClient client)
+        {
+            #region Get All Permissions
+
+            //*********************************************************************************************
+            // get all Permission Objects
+            //*********************************************************************************************
+            Console.WriteLine("\n Getting Permissions");
+            IPagedCollection<IOAuth2PermissionGrant> permissions = null;
+            try
+            {
+                permissions = await client.Oauth2PermissionGrants.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Getting Permissions: {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+            if (permissions != null)
+            {
+                do
+                {
```

Comment:
still see if followed by do-while

---
In reply to: [81021321](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r81021321) [](ancestors = 81021321,80756864)